node/src/main/scala/org/bitcoins/node/SpvNode.scala

```diff
@@ -50,30 +83,35 @@ case class SpvNode(
     * with P2P messages, therefore marked as
     * `private[node]`.
     */
-  private[node] def send(msg: NetworkPayload): Unit = {
-    peerMsgSender.sendMsg(msg)
+  private[node] def send(msg: NetworkPayload): Future[Unit] = {
+    peerMsgSenderF.map(_.sendMsg(msg))
   }
 
   /** Starts our spv node */
   def start(): Future[SpvNode] = {
     for {
       _ <- nodeAppConfig.initialize()
       node <- {
-        peerMsgSender.connect()
+        val connectedF = peerMsgSenderF.map(_.connect())
 
-        val isInitializedF =
-          AsyncUtil.retryUntilSatisfied(peerMsgRecv.isInitialized)
+        val isInitializedF = for {
+          _ <- connectedF
```

Comment:
```suggestion
          _ <- peerMsgSenderF.map(_.connect())
```

No real need for the intermediate val